lib/chef/resource/windows_share.rb

```diff
@@ -338,8 +335,38 @@ def bool_string(bool)
           # bool ? 1 : 0
           bool ? "$true" : "$false"
         end
+
+        # Grant-SmbShareAccess sets only one permission to one user so we need to remove from the lower permission access
+        # For change_users remove common full_users from it
+        # For read_users remove common full_users as well as change_users
+        def new_resource_users
+          users = {}
+          users["full_users"] = new_resource.full_users
+
+          users["change_users"] = new_resource.change_users - new_resource.full_users
+          users["read_users"] = new_resource.read_users - (new_resource.full_users + new_resource.change_users)
```

Comment:
this hash indirection should be removed and these should just be plain ruby variables in the method below.

Reply:
@lamont-granquist, i have also used this method in other methods too e.g. https://github.com/MsysTechnologiesllc/chef/blob/3c268b8c94db20d64ae729ce0525a8951272c8b9/lib/chef/resource/windows_share.rb#L313

Reply:
yeah but those accesses should change as well, these should just be `@full_users` ivars with accessors on them rather than doing access to the state through an unnecessary Hash.

Reply:
ok will change accordingly, Thanks!! @lamont-granquist